crates/daphne/src/lib.rs

```diff
@@ -732,7 +732,7 @@ impl DapTaskConfig {
             DapQueryConfig::FixedSize {
                 max_batch_size: Some(max_batch_size),
             } => {
-                if report_count > max_batch_size {
+                if (report_count > u64::from(max_batch_size)) && max_batch_size != 0 {
```

Comment:
why are you checking that max_batch_size != 0 here?

Reply:
Per the old draft a `max_batch_size` of 0 means there is no limit.
```
The maximum batch size for fixed_size query is optional. If query_type is fixed_size and max_batch_size is 0, then the task does not have maximum batch size limit.
```
https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-taskprov-00#section-3.1-7